workflow/controller/workflowpod.go

```diff
@@ -376,7 +376,17 @@ func (woc *wfOperationCtx) createWorkflowPod(ctx context.Context, nodeName strin
 		podSpecPatchs = append(podSpecPatchs, processedTmpl.PodSpecPatch)
 	}
 	if tmpl.HasPodSpecPatch() {
-		podSpecPatchs = append(podSpecPatchs, tmpl.PodSpecPatch)
+		localParams := make(map[string]string)
```

Comment:
This is duplicated code, can you tidy it into a function?

Reply:
Addressed!